src/server/master_server/master_chunk_server_manager_service_impl.cc

```diff
@@ -63,6 +66,24 @@ grpc::Status MasterChunkServerManagerServiceImpl::ReportChunkServer(
     // on it.
     absl::flat_hash_set<std::string> chunks_to_remove;
 
+    // Check for deleted chunk with metadata mgr. If a chunk handle is found
+    // deleted, we add it to stale chunks in reply so chunk server can delete
+    // them.
+    // TODO(Xi/bmokutub): Not sure if this piece of logic should / need to be 
+    // combined with below. TBD
+    for(auto stored_chunk_handles : 
+            request->chunk_server().stored_chunk_handles()) {
+      if (!gfs::server::MetadataManager::GetInstance()->ExistFileChunkMetadata(
+              stored_chunk_handles)) {
+         LOG(INFO) << "Chunk handle " << stored_chunk_handles 
+                   << " no longer existed in the master's metadata,"
+                   << " marking as staled chunk to chunk server "
+                   << new_server_info.DebugString();
+         reply->mutable_stale_chunk_handles()->Add(
+             std::move(stored_chunk_handles));
```

Comment:
Please also remove this chunk_handle from the `chunks_to_add` set created above to avoid adding it to chunkservermgr. Can do that here.

Reply:
done.